src/Presentation/Desktop/Pango.Desktop.Uwp/ViewModels/PasswordsViewModel.cs

```diff
@@ -534,6 +535,54 @@ private bool RemovePassword(ObservableCollection<PangoExplorerItem> passwords, P
         }
     }
 
+    /// <summary>
+    /// Saves all <see cref="Passwords"/> in the current state
+    /// </summary>
+    public async Task UpdatePasswordItemsInTreeAsync()
```

Comment:
This method doen't look efficient: 
1. I think, it would be better to have both lists prepared once data loaded instead of traversing the tree every time once a password dropped
2. You're casting to PangoPasswordListItemDto but it's not needed as I see